src/common/tse.c

```diff
@@ -466,10 +473,12 @@ static bool
 tse_task_complete_callback(tse_task_t *task)
 {
 	struct tse_task_private	*dtp = tse_task2priv(task);
+	struct tse_sched_private *dsp = dtp->dtp_sched;
 	uint32_t		 dep_cnt = dtp->dtp_dep_cnt;
 	struct tse_task_cb	*dtc;
 	struct tse_task_cb	*tmp;
 
+	D_MUTEX_LOCK(&dsp->dsp_comp_lock);
```

Comment:
Just curious, why not use dsp_lock?

Reply:
to avoid some possible dead lock.
BTW, I don't really know why need this patch. But it fixed a dfuse multi-threads testing issue...
I just want to fix the 1.2 blocker firstly, and can see more for it later